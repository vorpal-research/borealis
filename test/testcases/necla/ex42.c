extern int __NONDET__();
  
char x[101], y[101], z[201];
int from,to,i,j,k;

void main(){
  from = __NONDET__();
  to = __NONDET__();
  k = __NONDET__();

  if(!(k >=0 && k <= 100 && x[k] == 0))  /* assume strlen(x) <= 100 */
  {stuck1: goto stuck1;}

  if(!(from >= 0 && from <= k))            /* assume "from" index is O.K. */
  {stuck1: goto stuck1;}
  
  /* extract substring form index "from" to index "to" */
  
  i = from;
  j = 0;
  while(x[i] != 0 && i < to){
     z[j] = x[i];
     i++;
     j++;
  }
  
  z[j] = 0;
  
  if(!(j <= 100))
  {ERROR: goto ERROR;}  /* prove we don't overflow z */
}
