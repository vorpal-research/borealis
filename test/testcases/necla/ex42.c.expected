
[
	
	{
		"location" : 
		{
			"filename" : "test/testcases/necla/ex42.c",
			"loc" : 
			{
				"col" : 3,
				"line" : 21
			}
		},
		"type" : "BUF-01"
	},
	
	{
		"location" : 
		{
			"filename" : "test/testcases/necla/ex42.c",
			"loc" : 
			{
				"col" : 6,
				"line" : 22
			}
		},
		"type" : "BUF-01"
	}
]
